File: leetCode/1029. 两地调度/Main.py
class Solution(object):
    def twoCitySchedCost(self, costs):
        """
        :type costs: List[List[int]]
        :rtype: int
        """
        diff = map(lambda x: x[1]-x[0], costs)
        ans = sum(map(lambda x: x[0], costs))
        diff.sort()
        n = len(diff) / 2
        for i in range(n):
            ans += diff[i]
        return ans

File: leetCode/1029. 两地调度/Main.java
class Solution {
    public int twoCitySchedCost(int[][] costs) {
        int N = costs.length;
        int total = 0;
        for (int i = 0; i < N; i++) {
            total += costs[i][0];
        }
        Arrays.sort(costs, (a, b) -> ((a[1] - a[0]) - (b[1] - b[0])));
        for (int i = 0; i < N / 2; i++) {
            total += (costs[i][1] - costs[i][0]);
        }
        return total;
    }
}
